Content.Server/GameObjects/Components/Nutrition/SmokingComponent.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Threading.Tasks;
+using Content.Server.GameObjects.Components.GUI;
+using Content.Server.GameObjects.Components.Items.Clothing;
+using Content.Server.GameObjects.Components.Items.Storage;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.GameObjects.Components.Inventory;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components.Timers;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Nutrition
+{
+    /// <summary>
+    /// This item acts as a representation for smokable consumables.
+    ///
+    /// To smoke a cigar, you need:
+    /// <list type="bullet">
+    /// <item><description> a hot item (implements IHotItem interface)</description></item>
+    /// <item><description> that's a alight.</description></item>
+    /// <item><description>  for the target cigar be Unlit. Lit cigars are already lit and butt's don't have any "fuel" left.</description></item>
+    ///</list>
+    /// TODO: Add reagents that interact when smoking
+    /// TODO: Allow suicide via excessive Smoking
+    /// </summary>
+    [RegisterComponent]
+    public class SmokingComponent : Component, IInteractUsing, IHotItem
+    {
+        public override string Name => "Smoking";
+
+        private SharedBurningStates _currentState;
```

Comment:
could also be initialized here
```suggestion
        private SharedBurningStates _currentState = SharedBurningStates.Unlit;
```